lib/TileRenderer.js

```diff
@@ -215,15 +200,35 @@ class TileRenderer {
         // Draw the rectangle.
         gl.drawArrays(gl.TRIANGLES, 0, 6)
 
-        return cloneCanvas(canvasElement)
+        return cloneCanvas(canvasElement, createCanvas)
+    }
+}
+
+const createCanvasFromCreator = (width, height, creator = null) => {
```

Comment:
We're using the convention of `factory` to indicate a factory function elsewhere (rather than `creator`).